sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -66,14 +68,24 @@ public LogsQueryClient(Uri endpoint, TokenCredential credential) : this(endpoint
         /// <param name="endpoint">The service endpoint to use.</param>
         /// <param name="credential">The <see cref="TokenCredential"/> instance to use for authentication.</param>
         /// <param name="options">The <see cref="LogsQueryClientOptions"/> instance to use as client configuration.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public LogsQueryClient(Uri endpoint, TokenCredential credential, LogsQueryClientOptions options)
         {
             Argument.AssertNotNull(credential, nameof(credential));
             Argument.AssertNotNull(endpoint, nameof(endpoint));
 
             Endpoint = endpoint;
             options ??= new LogsQueryClientOptions();
-            var authorizationScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? LogsQueryAudience.AzurePublicCloud : options.Audience)}";
+            if (string.IsNullOrEmpty(options.Audience?.ToString()))
+            {
+                options.Audience = endpoint.OriginalString;
+            }
+            else if (endpoint.Host != new Uri(options.Audience.ToString()).Host)
+            {
+                throw new InvalidOperationException("The endpoint URI and audience do not match. If setting the Audience which is regionally specific, please use the LogsQueryClient(TokenCredential, LogsQueryClientOptions) constructor.");
```

Comment:
@scottaddie Could you please check if this is clear? I'm trying to provide as much context as possible and we have also suppressed the constructors that include the Uri.

Reply:
What was the previous behavior here if a user passed in a host that didn't match the audience?

Reply:
We didn't have code for this scenario before :(. If the user didn't pass in Options we automatically set it to public cloud which breaks if your endpoint is in a different region.

Reply:
But what would happen if the user _did_ pass in options but it contained an audience different from the endpoint? I ask because now we are throwing, but I'm wondering if the behavior previously would have been acceptable for users and therefore this change might break them.

Reply:
I don't think we will break any customers because the code does not run if the Endpoint and Audience do not match. Previously, the service threw an exception when this happened, I'm just adding the validation on the client side.